### DISCUSSION TITLE
is it possible to make an extension for agents? like auto gpt or babyagi?

### DISCUSSION THREAD

**Discussion by yesbroc** [oobabooga/text-generation-webui#1793] [Ideas]
a custom version of babyagi maybe since it already supports both llama.cpp and langchain. or imitate openai's api and run auto gpt

**Comment by eucdee** [comment]
well those agents can talk to ooba via the api,
check out [agent-llm](https://github.com/Josh-XT/Agent-LLM) 
great dev i need to keep up, havent check the repo for a couple of days now.
to be fair you can also check out other [babyagi inspired projects](https://github.com/yoheinakajima/babyagi/blob/main/docs/inspired-projects.md)

so what would be the idea here? a button to quickly deploy an agent?
maybe the complete frontend of the agent inside the web-ui?
for the web-ui to have its own agents?

it certainly is possible to awnser your question :)

**Comment by Felixqian4160** [comment]
it already done
https://github.com/flurb18/babyagi4all-api

**Reply by yesbroc** [reply]
i honestly dont like babyagi since it takes a while to complete and sets its own tasks. autogpt lets you set 5 tasks manually which i love
